template_reports/office_renderer/loops.py

```diff
@@ -0,0 +1,235 @@
+"""
+Functions to handle loop functionality in PPTX templates.
+"""
+
+import re
+from typing import Dict, List, Optional, Tuple, Any
+from copy import deepcopy
+
+from ..templating import resolve_tag
+
+# Patterns for loop directives
+LOOP_START_PATTERN = re.compile(r"%\s*loop\s+(\w+)\s+in\s+(.+?)\s*%")
```

Comment:
let's create a new constants file somewhere to hold import key words like "loop" and "in" and "endloop"

pull in other key directive words too (eg "image" and "imagesqueeze" from the image insertion code)